app/src/main/java/com/example/taskyy/ui/AgendaScreen.kt

```diff
@@ -1,12 +1,203 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import com.example.taskyy.R
+import com.example.taskyy.domain.navigation.Screen
+import com.example.taskyy.ui.events.AgendaEvent
+import com.example.taskyy.ui.viewmodels.AgendaState
+import java.time.LocalDateTime
+import java.time.ZoneId
 
 @Composable
-fun AgendaScreen(){
-    Column {
-        Text(text = "Agenda Screen")
+fun AgendaScreen(state: AgendaState, navController: NavController, onEvent: (AgendaEvent) -> Unit){
+    Column(
+        verticalArrangement = Arrangement.SpaceAround,
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color.Black)
+    ) {
+        Row (
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = 10.dp)
+        ) {
+            Row(
+                horizontalArrangement = Arrangement.Start,
+                modifier = Modifier.clickable {
+                    onEvent(AgendaEvent.OnMonthExpanded(!state.isMonthExpanded))
+                }
+            ) {
+                MonthText(
+                    selectedMonth = state.selectedMonth
+                )
+                MonthDownArrow()
+                if (state.isMonthExpanded) {
+                    ShowDatePicker(
+                        onItemSelected = { selectedMonth ->
+                            onEvent(AgendaEvent.OnDateSelected(selectedMonth))
+                        },
+                    )
+                }
+            }
+            CircleWithInitials(
+                isUserDropDownExpanded = state.isUserDropDownExpanded,
+                onUserInitialsClicked = { isUserDropDownExpanded ->
+                    onEvent(AgendaEvent.OnUserInitialsClicked(isUserDropDownExpanded))
+                },
+                onItemSelected = {
+                    onEvent(AgendaEvent.OnLogOutCLicked)
+                },
+            )
+        }
+        Surface(
+            modifier = Modifier
+                .fillMaxHeight(.85F)
+                .fillMaxWidth(),
+            shape = AbsoluteRoundedCornerShape(30.dp, 30.dp)
+        ) {
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.Top,
+                modifier = Modifier
+                    .fillMaxHeight(.75f)
+                    .padding(30.dp)
+            ) {
+                Spacer(modifier = Modifier.height(10.dp))
+            }
+        }
     }
+    if(state.wasLogoutSuccessful){
+        navController.navigate(Screen.Login.route)
+    }
```

Comment:
Something like this is dangerous - we call that a side-effect. Since you have no control over when and how often a composable function is called, putting non-Compose functions (like `navController.navigate`) inside a composable function can lead to very weird behavior. In this case it's unlikely going to lead to an issue because `state.wasLogoutSuccessful` is false for most of the time, but in general, you'd want to handle these with Compose's effect handlers. In this case like this:
```
// This block will trigger every time the passed state changes
LaunchedEffect(state.wasLogoutSuccessful) {
    if(state.wasLogoutSuccessful) {
        navController.navigate(Screen.Login.route) {
            popUpTo(Screen.Agenda.route) { // This is important, since otherwise the user can get back to the agenda screen after logging out
                inclusive = true 
            }
        }
    }
}
```

Reply:
If I do not pass the navController into the agenda screen and I am handling navigation from the nav host, would I put this in the navhost since the agenda screen composable wouldn't have a navController reference?

Reply:
Yes, or in the NavHost you could simply do what I described in another comment:
```
composable("agenda) {
    AgendaScreen(
        state = viewModel.state,
        onEvent = { event ->
            when(event) {
                is AgendaEvent.Logout -> // navigate to login
                is AgendaEvent.OnAgendaItemClick -> // navigate to agenda details
                ...
            }
            viewModel.onEvent(event) // Still send all events to the VM so nothing is lost
        }
    )
}
```

Reply:
When I implement it this way, it no longer validates the log in. It just logs the user in. I added

`is AgendaEvent.Login -> if(state.isLoginSuccessful) {
          navController.navigate(Screen.Agenda.route)
          }`

This makes me have to click login twice since the state isn't updated immediately on validation.

Reply:
In how far does the agenda feature have something to do with logging in? 🤔